nncf/torch/quantization/reference.py

```diff
@@ -119,7 +120,11 @@ def tune_range(
         return new_input_low, new_input_range
 
 
+torch_executor = ReferenceQuantize(backend_type=ReferenceBackendType.TORCH)
+torch_forward = CompilationWrapper(torch_executor.forward)
+torch_backward = CompilationWrapper(torch_executor.backward)
```

Comment:
Wrapped function lose annotation 
![image](https://github.com/user-attachments/assets/348e114a-08ae-4b55-93ba-467510cd53ad)
![image](https://github.com/user-attachments/assets/436ddccc-684f-443a-b009-dbbff88d6430)

If write decorator like function, correctly it looks like https://github.com/openvinotoolkit/nncf/blob/dd99ed548e895a5625a33e53a762c055dd0df825/nncf/common/utils/caching.py#L61
(used functools.wrap and TypeVar to keep signature and docstring of function)
It breaks suggestion in editors and check arguments by mypy, but i dont know how do it for class.